### PR TITLE
Migrate odometry display

### LINKED ISSUE #80: Migrate OdometryDisplay

### PR DESCRIPTION
**Pull request by Martin-Idel-SI**
Closes #80 
- also ports covariance visual and covariance property as it is used in odometry display
- makes covariance property and visual work like other visuals and properties
- further improve Visual Testing Framework: can now easily search for subproperties

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4519/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1438/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3710/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4596/)

### CODE REVIEW

**Comment by Martin-Idel-SI** [comment]
I have no idea why this always fails - it's not rviz. I updated my ros2.repos file, but that's apparently not it.

**Comment by mikaelarguedas** [comment]
It looks like the repos file used in the CI above is not exactly up to date, as it doesnt contain: https://github.com/ros2/ros2/pull/491. That's why CI is failing complaining about a missing `libyaml_vendor` package

<details>

```
--- stderr: rcl_yaml_param_parser
00:07:40.622 CMake Error at CMakeLists.txt:8 (find_package):
00:07:40.622   By not providing "Findlibyaml_vendor.cmake" in CMAKE_MODULE_PATH this
00:07:40.622   project has asked CMake to find a package configuration file provided by
00:07:40.622   "libyaml_vendor", but CMake did not find one.
00:07:40.622 
00:07:40.622   Could not find a package configuration file provided by "libyaml_vendor"
00:07:40.622   with any of the following names:
00:07:40.622 
00:07:40.622     libyaml_vendorConfig.cmake
00:07:40.622     libyaml_vendor-config.cmake
00:07:40.622 
00:07:40.622   Add the installation prefix of "libyaml_vendor" to CMAKE_PREFIX_PATH or set
00:07:40.622   "libyaml_vendor_DIR" to a directory containing one of the above files.  If
00:07:40.623   "libyaml_vendor" provides a separate development package or SDK, be sure it
00:07:40.623   has been installed.
00:07:40.623 
00:07:40.623 
00:07:40.623 ---

```

**Comment by Martin-Idel-SI** [comment]
Weird, because I did update it just today... I guess I must've forgotten to enter the new URL before rebuilding... Thanks for looking into this!

Here's hopefully better CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4528/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1446/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3719/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4605/)

**Comment by botteroa-si** [comment]
New CI after rebase: 

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4622/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1522/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3809/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4700/)

File: rviz_visual_testing_framework/src/page_objects/base_page_object.cpp [hunk 1]
```diff
@@ -113,96 +101,93 @@ void BasePageObject::setComboBox(const QString & property_name, const QString &
...
-    [this, value_to_set, main_property_name, sub_property_index, sub_property_name] {
-      auto relative_display_index = getRelativeIndexAndExpandDisplay();
-      int property_row_index =
-      findPropertyRowIndexByName(main_property_name, relative_display_index);
-
-      auto value_to_change_index = sub_property_index < 0 ?
-      getMainPropertyIndex(main_property_name, property_row_index, relative_display_index) :
-      getSubPropertyIndex(
-        sub_property_name, property_row_index, sub_property_index, relative_display_index);
+    [this, value_to_set, property_to_change, super_properties{std::move(super_properties)}] {
```
Comment (wjwwood) [inline rviz_visual_testing_framework/src/page_objects/base_page_object.cpp:109]: I like this use of initializer list within a lambda capture statement. 🆒

**Review by wjwwood — APPROVED** [review]

**Pull request merged**